operators.go

```diff
@@ -1,42 +1,53 @@
 package plugin
 
 import (
+	"fmt"
 	"net"
-	"strconv"
 
-	"github.com/corazawaf/coraza/v2"
-	"github.com/corazawaf/coraza/v2/types/variables"
-	geoip "github.com/oschwald/geoip2-golang"
+	"github.com/corazawaf/coraza/v3/collection"
+	"github.com/corazawaf/coraza/v3/experimental/plugins/plugintypes"
+	"github.com/corazawaf/coraza/v3/types/variables"
+	"github.com/oschwald/geoip2-golang"
 )
 
-type geoLookup struct{}
+// this acts as a shim for the geoip2.Reader struct so we can mock it in tests
+type geoIPReaderShim interface {
```

Comment:
Do we really need the Shim suffix? can we just use `geoIPReader`, same for others.